### PR TITLE
Create a webhook to disable bounced emails

### PR DESCRIPTION
**Pull request by JoaoVictor6**
In this PR I implement a webhook to set the `opt-out` column as `true` when sending the email if the email was bounced.

This implementation use [svix](https://www.npmjs.com/package/svix) to verify webhook signature to avoid security problems.

### CODE REVIEW

File: apps/web/app/(resend_webhooks)/api/webhooks/bounce/route.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    const { error } = await supabaseClient
+      .from('Subscribers')
+      .update({ optOut: true })
+      .eq('email', bouncedEmail)
+    if (error) {
+      console.error('Error when update bounced email: ', error)
+    }
+  }
+
+  return new Response()
```
Comment (ocodista) [inline apps/web/app/(resend_webhooks)/api/webhooks/bounce/route.ts:57]: According to the Vercel document, we should return 200 if the webhook is successful, does this Response returns 200?

**Review by ocodista — CHANGES_REQUESTED** [review]

File: apps/web/app/(resend_webhooks)/api/webhooks/bounce/route.ts (hunk 1 above)
Comment (JoaoVictor6) [reply apps/web/app/(resend_webhooks)/api/webhooks/bounce/route.ts:57]: by default yes. I will explicit this in code.

**Review by ocodista — APPROVED** [review]

File: apps/web/app/(resend_webhooks)/api/webhooks/bounce/route.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  }) as WebhookEvent
+
+  if (payload.type === 'email.bounced') {
+    const supabaseClient = getSupabaseClient()
+    const bouncedEmail = payload.data.to[0]
+
+    const { error } = await supabaseClient
+      .from('Subscribers')
+      .update({ optOut: true })
+      .eq('email', bouncedEmail)
```
Comment (ocodista) [inline apps/web/app/(resend_webhooks)/api/webhooks/bounce/route.ts:51]: We should also update the `updatedAt` field

**Review by ocodista — CHANGES_REQUESTED** [review]

File: apps/web/app/(resend_webhooks)/api/webhooks/bounce/route.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  const svix_id = req.headers.get('svix-id') ?? ''
+  const svix_timestamp = req.headers.get('svix-timestamp') ?? ''
+  const svix_signature = req.headers.get('svix-signature') ?? ''
+
+  const body = await req.text()
+  const payload = webhook.verify(body, {
+    'svix-id': svix_id,
+    'svix-timestamp': svix_timestamp,
+    'svix-signature': svix_signature,
+  }) as WebhookEvent
```
Comment (ocodista) [inline apps/web/app/(resend_webhooks)/api/webhooks/bounce/route.ts:42]: What happens if this verification fails?

We don't have a try catch here... We should return 500

**Review by ocodista — CHANGES_REQUESTED** [review]

File: apps/web/app/(resend_webhooks)/api/webhooks/bounce/route.ts (hunk 3 above)
Comment (JoaoVictor6) [reply apps/web/app/(resend_webhooks)/api/webhooks/bounce/route.ts:42]: The SVIX validates the header and throws an error if it is not valid and sends a status 500.

I created an example without necessary headers and the svix thow this error:
image

**Pull request merged**